standard/expressions.md

```diff
@@ -4709,28 +4784,30 @@ The tuple equality operator `x != y` is evaluated as follows:
 
 There are two forms of the `is` operator. One is the *is-type operator*, which has a type on the right-hand-side. The other is the *is-pattern operator*, which has a pattern on the right-hand-side.
 
+An expression of the form `E is _`, where the right-hand-side is the identifier `_` and designates a type, is an *is-type operator*. To reduce confusion with the discard pattern, an implementation shall issue a warning where this occurs.
+
 #### 12.13.12.1 The is-type operator
 
 The *is-type operator* is used to check if the run-time type of an object is compatible with a given type. The check is performed at runtime. The result of the operation `E is T`, where `E` is an expression and `T` is a type other than `dynamic`, is a Boolean value indicating whether `E` is non-null and can successfully be converted to type `T` by a reference conversion, a boxing conversion, an unboxing conversion, a wrapping conversion, or an unwrapping conversion.
 
-The operation is evaluated as follows:
+The operation `E is T` is evaluated as follows:
 
 1. If `E` is an anonymous function or method group, a compile-time error occurs.
+1. If `T` is a nullable reference type ([§8.9.3](types.md#893-nullable-reference-types)), a compile-time error occurs.
 1. If `E` is the `null` literal, or if the value of `E` is `null`, the result is `false`.
 1. Otherwise:
-1. Let `R` be the runtime type of `E`.
-1. Let `D` be derived from `R` as follows:
-1. If `R` is a nullable value type, `D` is the underlying type of `R`.
-1. Otherwise, `D` is `R`.
-1. The result depends on `D` and `T` as follows:
-1. If `T` is a reference type, the result is `true` if:
-    - an identity conversion exists between `D` and `T`,
-    - `D` is a reference type and an implicit reference conversion from `D` to `T` exists, or
-    - Either: `D` is a value type and a boxing conversion from `D` to `T` exists.  
-      Or: `D` is a value type and `T` is an interface type implemented by `D`.
-1. If `T` is a nullable value type, the result is `true` if `D` is the underlying type of `T`.
-1. If `T` is a non-nullable value type, the result is `true` if `D` and `T` are the same type.
-1. Otherwise, the result is `false`.
+  1. Let `R` be the runtime type of `E`.
+  1. Let `D` be derived from `R` as follows:
+    1. If `R` is a nullable value type, `D` is the underlying type of `R`.
+    1. Otherwise, `D` is `R`.
```

Comment:
Needs one more indent space:

```suggestion
     1. If `R` is a nullable value type, `D` is the underlying type of `R`.
     1. Otherwise, `D` is `R`.
```

Reply:
I think it needs more than one as the preceding lines are also being barfed at, isn’t MD “fun”!